### PR TITLE
refactor(connector): remove default cases for Authorizedotnet, Braintree and Fiserv Connector

### PR DESCRIPTION
**Pull request by swangi-kumari**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Resolves [4376](https://github.com/juspay/hyperswitch-cloud/issues/4376)
Issue: defaults case in match shouldn't be handled using "_"

Solution: "_" will never let the developer know impact of his changes incase of new addition of enum variants. By having the all cases in match compiler itself will throw errors wherever new enums needs to be handled.

(In this PR it is done for Authorizrdotnet, Braintree and Fiserv Connector)

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Do payment create for Authorizedotnet, Braintree and Fiserv Connector for any payment method which is not implemented it should throw the error 

here is payment request for PIX PM 
```json
{
    "amount": 6540,
    "currency": "BRL",
    "confirm": true,
    "capture_method": "automatic",
    "capture_on": "2022-09-10T10:11:12Z",
    "amount_to_capture": 6540,
    "customer_id": "StripeCustomer",
    "email": "guest@example.com",
    "name": "John Doe",
    "phone": "999999999",
    "phone_country_code": "+1",
    "description": "Its my first payment request",
    "authentication_type": "no_three_ds",
    "return_url": "https://duck.com",
     "payment_method": "bank_transfer",
    "payment_method_type": "pix",
    "payment_method_data": {
        "bank_transfer": {
            "pix": {}
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "BR",
            "first_name": "PiX"
        }
    },
    "shipping": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "BR",
            "first_name": "PiX"
        }
    },
    "statement_descriptor_name": "joseph",
    "statement_descriptor_suffix": "JS",
    "metadata": {
        "udf1": "value1",
        "new_customer": "true",
        "login_date": "2019-09-10T10:11:12Z"
    }
}
```

```txt
Selected payment method is Not Implemented for <connector-name>
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by Sakilmostak — DISMISSED** [review]

**Review by Sakilmostak — APPROVED** [review]

File: crates/router/src/connector/fiserv/transformers.rs [hunk 1]
```diff
@@ -182,9 +182,25 @@ impl TryFrom<&FiservRouterData<&types::PaymentsAuthorizeRouterData>> for FiservP
...
+            | api::PaymentMethodData::BankTransfer(_)
+            | api::PaymentMethodData::Crypto(_)
+            | api::PaymentMethodData::MandatePayment
+            | api::PaymentMethodData::Reward
+            | api::PaymentMethodData::Upi(_)
+            | api::PaymentMethodData::Voucher(_)
+            | api::PaymentMethodData::GiftCard(_)
+            | api::PaymentMethodData::CardToken(_) => {
+                Err(errors::ConnectorError::NotImplemented(
+                    utils::get_unimplemented_payment_method_error_message("braintree"),
```
Comment (srujanchikke) [inline crates/router/src/connector/fiserv/transformers.rs:201]: This is supposed to be `fiserv` .

**Review by srujanchikke — CHANGES_REQUESTED** [review]

**Review by srujanchikke — DISMISSED** [review]

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 2]
```diff
@@ -295,7 +318,9 @@ impl From<enums::CaptureMethod> for AuthorizationType {
     fn from(item: enums::CaptureMethod) -> Self {
         match item {
             enums::CaptureMethod::Manual => Self::Pre,
-            _ => Self::Final,
+            euclid::enums::CaptureMethod::Automatic
+            | enums::CaptureMethod::ManualMultiple
+            | enums::CaptureMethod::Scheduled => Self::Final,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/authorizedotnet/transformers.rs:323]: ManualMultiple and Scheduled is not supported by the connector. In such cases throw error

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 3]
```diff
@@ -904,7 +929,14 @@ impl From<SyncStatus> for enums::RefundStatus {
...
             SyncStatus::RefundPendingSettlement => Self::Pending,
-            _ => Self::Failure,
+            SyncStatus::AuthorizedPendingCapture
+            | SyncStatus::CapturedPendingSettlement
+            | SyncStatus::SettledSuccessfully
+            | SyncStatus::Declined
+            | SyncStatus::Voided
+            | SyncStatus::CouldNotVoid
+            | SyncStatus::GeneralError
+            | SyncStatus::FDSPendingReview => Self::Failure,
```
Comment (ArjunKarthik) [inline crates/router/src/connector/authorizedotnet/transformers.rs:939]: Why FDSPendingReview, SettledSuccessfully, AuthorizedPendingCapture, CapturedPendingSettlement, Voided should be mapped to Failure?

File: crates/router/src/connector/braintree/transformers.rs [hunk 4]
```diff
@@ -129,14 +129,62 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for BraintreePaymentsRequest {
...
                             Ok(wallet_data.token.to_owned())
                         }
-                        _ => Err(errors::ConnectorError::InvalidWallet),
+                        api_models::payments::WalletData::ApplePay(_)
+                        | api_models::payments::WalletData::GooglePay(_)
+                        | api_models::payments::WalletData::SamsungPay(_) => {
+                            Err(errors::ConnectorError::NotImplemented(
+                                utils::get_unimplemented_payment_method_error_message("braintree"),
+                            ))
+                        }
```
Comment (ArjunKarthik) [inline crates/router/src/connector/braintree/transformers.rs:138]: why are we throwing the same error in to places instead use | ?

File: crates/router/src/connector/braintree/transformers.rs [hunk 5]
```diff
@@ -129,14 +129,62 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for BraintreePaymentsRequest {
...
             }
-            _ => Err(errors::ConnectorError::NotImplemented(format!(
-                "Current Payment Method - {:?}",
-                item.request.payment_method_data
-            ))),
+            api::PaymentMethodData::PayLater(_)
+            | api::PaymentMethodData::BankRedirect(_)
+            | api::PaymentMethodData::BankDebit(_)
+            | api::PaymentMethodData::BankTransfer(_) => {
+                Err(errors::ConnectorError::NotImplemented(
```
Comment (ArjunKarthik) [inline crates/router/src/connector/braintree/transformers.rs:172]: why are we throwing the same error in to places instead use | ?

File: crates/router/src/connector/fiserv/transformers.rs [hunk 6]
```diff
@@ -182,9 +182,25 @@ impl TryFrom<&FiservRouterData<&types::PaymentsAuthorizeRouterData>> for FiservP
                 };
                 Source::PaymentCard { card }
             }
-            _ => Err(errors::ConnectorError::NotImplemented(
-                "Payment Methods".to_string(),
+            api::PaymentMethodData::Wallet(_)
+            | api::PaymentMethodData::PayLater(_)
+            | api::PaymentMethodData::BankRedirect(_)
+            | api::PaymentMethodData::BankDebit(_) => Err(errors::ConnectorError::NotImplemented(
+                utils::get_unimplemented_payment_method_error_message("fiserv"),
```
Comment (ArjunKarthik) [inline crates/router/src/connector/fiserv/transformers.rs:189]: Same here

**Review by ArjunKarthik — APPROVED** [review]

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]
Changes required

File: crates/router/src/connector/authorizedotnet/transformers.rs (hunk 3 above)
Comment (swangi-kumari) [reply crates/router/src/connector/authorizedotnet/transformers.rs]: @deepanshu-iiitu  will handle the status mapping modifications in the pull request where he is addressing the connector audit fixes.

File: crates/router/src/connector/authorizedotnet/transformers.rs [hunk 7]
```diff
@@ -307,13 +335,13 @@ impl TryFrom<&AuthorizedotnetRouterData<&types::PaymentsAuthorizeRouterData>>
...
         let (payment_details, processing_options, subsequent_auth_information) =
             get_pm_and_subsequent_auth_detail(item)?;
-        let authorization_indicator_type =
-            item.router_data
-                .request
-                .capture_method
-                .map(|c| AuthorizationIndicator {
-                    authorization_indicator: c.into(),
-                });
+        let authorization_indicator_type = match item.router_data.request.capture_method {
```
Comment (ArjunKarthik) [inline crates/router/src/connector/authorizedotnet/transformers.rs:338]: Why do we need this change?

Comment (swangi-kumari) [reply crates/router/src/connector/authorizedotnet/transformers.rs:338]: I refactored the CaptureMethod from `from` to `try_from` , in order to throw the error for `ManualMultiple` and `Scheduled`. this was a relative change for that

**Review by ArjunKarthik — APPROVED** [review]

**Review by deepanshu-iiitu — APPROVED** [review]

**Pull request merged**